### PR TITLE
Detection and conversion of InputStream with Byte Order Marks to UTF-8 encoded InputStream

### PR DESCRIPTION
**Pull request by flose89**

### CODE REVIEW

File: sapl-lang/src/main/java/io/sapl/interpreter/DefaultSAPLInterpreter.java [hunk 1]
```diff
@@ -69,14 +69,28 @@ public SAPL parse(String saplDefinition) {
...
     public SAPL parse(InputStream saplInputStream) {
-        var sapl       = loadAsResource(saplInputStream);
+
+        InputStream convertedSaplInputStream;
+        try {
+            convertedSaplInputStream = InputStreamHelper.detectAndConvertEncodingOfStream(saplInputStream);
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+            throw new PolicyEvaluationException(
+                    composeReason("Invalid byte sequence in InputStream. Could not transform to UTF-8"));
```
Comment (heutelbeck) [inline sapl-lang/src/main/java/io/sapl/interpreter/DefaultSAPLInterpreter.java:79]: Add cause exception to history

File: sapl-lang/src/main/java/io/sapl/interpreter/InputStreamHelper.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public static InputStream detectAndConvertEncodingOfStream(InputStream policyInputStream) throws IOException {
+
+        BOMInputStream bomIn = BOMInputStream
+                .builder().setInputStream(policyInputStream).setByteOrderMarks(ByteOrderMark.UTF_16LE,
+                        ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_32LE, ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_8)
+                .get();
+
+        if (!bomIn.hasBOM()) {
+            // InputStream without BOM is treated as UTF-8
+            return getUtf8InputStream(new InputStreamReader(bomIn, StandardCharsets.UTF_8));
```
Comment (heutelbeck) [inline sapl-lang/src/main/java/io/sapl/interpreter/InputStreamHelper.java:44]: See SonarCloud. Use Static default encoding Stings everywhere.

File: sapl-maven-plugin/src/test/java/io/sapl/mavenplugin/test/coverage/stubs/SaplMavenPluginTestMavenProjectStub.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public SaplMavenPluginTestMavenProjectStub() {
         MavenXpp3Reader pomReader = new MavenXpp3Reader();
         Model           model;
         try {
-            model = pomReader.read(new XmlStreamReader(new File(getBasedir(), "pom.xml")));
+            model = pomReader.read((new FileReader(new File(getBasedir(), "pom.xml"))));
```
Comment (heutelbeck) [inline sapl-maven-plugin/src/test/java/io/sapl/mavenplugin/test/coverage/stubs/SaplMavenPluginTestMavenProjectStub.java:42]: See Sonar. And why the change ?

**Review by heutelbeck — CHANGES_REQUESTED** [review]
Please address SonarCloud issues.

File: sapl-maven-plugin/src/test/java/io/sapl/mavenplugin/test/coverage/stubs/SaplMavenPluginTestMavenProjectStub.java (hunk 3 above)
Comment (flose89) [reply sapl-maven-plugin/src/test/java/io/sapl/mavenplugin/test/coverage/stubs/SaplMavenPluginTestMavenProjectStub.java]: I upgraded commons.io to version 2.15 and saw that the way XmlStreamReader is instantiated here is deprecated. I thought that doing without the class and thus without the import would be fine/even better.

File: sapl-lang/src/main/java/io/sapl/interpreter/InputStreamHelper.java (hunk 2 above)
Comment (flose89) [reply sapl-lang/src/main/java/io/sapl/interpreter/InputStreamHelper.java:44]: UTF-32 and similar are not part of the standard charset

https://bugs.openjdk.org/browse/JDK-8310251

Rest has been fixed.

File: sapl-lang/src/main/java/io/sapl/interpreter/DefaultSAPLInterpreter.java (hunk 1 above)
Comment (flose89) [reply sapl-lang/src/main/java/io/sapl/interpreter/DefaultSAPLInterpreter.java]: I hope that my updated PR resolves this issue. I'm unable to test it beforehand due to SonarLint issues.
